src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1422,10 +1425,145 @@ public override IOperation VisitMethodBodyOperation(IMethodBodyOperation operati
         {
             StartVisitingStatement(operation);
 
+            var member = operation.SemanticModel.GetDeclaredSymbol(operation.Syntax);
+            Debug.Assert(captureIdForResult is null);
+            VisitNullChecks(operation, ((IMethodSymbol)member).Parameters);
+
             VisitMethodBodyBaseOperation(operation);
             return FinishVisitingStatement(operation);
         }
 
+        private void VisitNullChecks(IOperation operation, ImmutableArray<IParameterSymbol> parameters)
+        {
+            var temp = _currentStatement;
+            foreach (var param in parameters)
+            {
+                if (param.IsNullChecked)
+                {
+                    var check = GenerateNullCheckForParameter(param, operation.Syntax, ((Operation)operation).OwningSemanticModel);
+                    _currentStatement = check;
+                    VisitConditional(check, captureIdForResult: null);
+                }
+            }
+            _currentStatement = temp;
+        }
+
+        private ConditionalOperation GenerateNullCheckForParameter(IParameterSymbol parameter, SyntaxNode syntax, SemanticModel semanticModel)
+        {
+            Debug.Assert(parameter.Language == LanguageNames.CSharp);
+            Optional<object> constantValue = default(Optional<object>);
+            var paramReference = new ParameterReferenceOperation(parameter, semanticModel, syntax, parameter.Type, constantValue, isImplicit: true);
+            var paramNameLiteral = new LiteralOperation(semanticModel, syntax, _compilation.GetSpecialType(SpecialType.System_String), parameter.Name, isImplicit: true);
+            var boolType = _compilation.GetSpecialType(SpecialType.System_Boolean);
+
+            IOperation conditionOp;
+            if (ITypeSymbolHelpers.IsNullableType(parameter.Type))
+            {
+                var nullableHasValue = ((IMethodSymbol)_compilation.CommonGetSpecialTypeMember(SpecialMember.System_Nullable_T_get_HasValue))?.Construct(parameter.Type);
+                if (nullableHasValue is null)
+                {
+                    conditionOp = new InvalidOperation(
+                        ImmutableArray.Create((IOperation)OperationCloner.CloneOperation(paramNameLiteral)),
```

Comment:
This should probably be the paramReference, not the paramNameLiteral. I would also still wrap it with a unary not, as it's the invocation that's bad, and not the not.